src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -341,11 +353,41 @@
     (->> (map format-task info-from-jruby)
          (middleware-utils/json-response 200))))
 
+(schema/defn ^:always-validate
+  describe-task :- TaskDetails
+  [task-data :- TaskData
+   env-name :- schema/Str
+   module-name :- schema/Str]
+  (let [?metadata (try (some-> task-data :metadata_file slurp json/decode)
+                   (catch Exception e
+                     nil))
+        describe-file (fn [file]
+                        (let [size (fs/size file)
+                              sha256 (ks/file->sha256 (io/file file))
+                              base-name (fs/base-name file)]
+                          {:filename base-name
+                           :sha256 sha256
+                           :size_bytes size
+                           :uri {:path (str "/puppet/v3/file_content/tasks/" module-name "/" base-name)
+                                 :params {:environment env-name}}}))]
+    {:metadata (or ?metadata {})
+     :files (mapv describe-file (:files task-data))}))
```

Comment:
why mapv?

Reply:
I tend to favor `mapv` when I know that I don't want laziness.